### PR TITLE
add types for @terraformer/arcgis

### PR DESCRIPTION
**Pull request by gavinr**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

https://github.com/terraformer-js/terraformer/issues/46 cc @jgravois

### CODE REVIEW

File: types/terraformer__arcgis/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+// Definitions by: Jeff Jacobson <https://github.com/JeffJacobson>, Gavin Rehkemper <https://github.com/gavinr>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.7
+
+import * as GeoJSON from 'geojson';
+import * as ArcGIS from 'arcgis-rest-api';
+
+// tslint:disable-next-line [no-unnecessary-generics]
+export function arcgisToGeoJSON(arcgis: ArcGIS.Geometry): GeoJSON.GeometryObject;
+export function geojsonToArcGIS(geojson: GeoJSON.GeometryObject): ArcGIS.Geometry;
```
Comment (SimplyLinn) [inline types/terraformer__arcgis/index.d.ts:12]:
```suggestion
export function arcgisToGeoJSON(arcgis: ArcGIS.Geometry, idAttribute?: string): GeoJSON.GeometryObject;
export function geojsonToArcGIS(geojson: GeoJSON.GeometryObject, idAttribute?: string): ArcGIS.Geometry;
```

**Review by SimplyLinn — CHANGES_REQUESTED** [review]
It appears that an optional idAttribute parameter is missing, which is specified in the documentation.

https://github.com/terraformer-js/terraformer/blob/master/packages/arcgis/README.md

File: types/terraformer__arcgis/index.d.ts (hunk 1 above)
Comment (gavinr) [reply types/terraformer__arcgis/index.d.ts]: Thanks a lot @SimplyLinn!

**Review by SimplyLinn — APPROVED** [review]
LGTM

File: types/terraformer__arcgis/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Type definitions for @terraformer/arcgis 2.0
+// Project: https://github.com/terraformer-js/terraformer
+// Definitions by: Jeff Jacobson <https://github.com/JeffJacobson>, Gavin Rehkemper <https://github.com/gavinr>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.7
+
+import * as GeoJSON from 'geojson';
+import * as ArcGIS from 'arcgis-rest-api';
+
+// tslint:disable-next-line [no-unnecessary-generics]
```
Comment (sandersn) [inline types/terraformer__arcgis/index.d.ts:10]: this isn't needed anymore, is it?

**Review by sandersn — CHANGES_REQUESTED** [review]

**Review by sandersn — APPROVED** [review]

File: types/terraformer__arcgis/index.d.ts (hunk 2 above)
Comment (gavinr) [reply types/terraformer__arcgis/index.d.ts]: 👍

**Comment by gavinr** [comment]
Ready to merge

**Pull request merged**